scripts/create_joined_view.py

```diff
@@ -396,7 +397,41 @@ def main(args):
                                     assignment_fields + assignment_type_fields+ worker_fields,
                                     dispatcher_fields)
     logger.info(f"Final Item: {final_item.title}")
-    logger.info("Completed")
+    if args.create_dashboard:
+        if gis.properties["isPortal"]:
+            raise ValueError("GIS needs to be AGOL for the --create-dashboard parameter to work")
+        logger.info("Creating dashboard")
+        
+        # create new webmap
+        map_item = project.dispatcher_webmap.save(
+            item_properties={"title": project.title + " Dashboard Map", "tags": [], "snippet": "Dashboard Map"})
+        new_webmap = WebMap(map_item)
+        
+        # swizzle in joined layer instead of assignments layer
+        for i, layer in enumerate(new_webmap.layers):
+            if layer["id"] == "Assignments_0":
+                new_webmap.layers[i]["itemId"] = final_item.id
```

Comment:
Is there a popup that needs to be configured to show the new joined fields?

Reply:
Good point @apulverizer I pushed a fix using `add_layer`